test/scripts/start-services.sh

```diff
@@ -113,36 +113,101 @@ start_polkadot_launch()
 
 start_relayer()
 {
-    echo "Starting Relay"
-    logfile=$(pwd)/relay.log
-    pushd ../relayer
-
-    mage build
-
-    export SNOWBRIDGE_BEEFY_KEY="0x935b65c833ced92c43ef9de6bff30703d941bd92a2637cb00cfad389f5862109"
-    export SNOWBRIDGE_MESSAGE_KEY="0x8013383de6e5a891e7754ae1ef5a21e7661f1fe67cd47ca8ebf4acd6de66879a"
-    export SNOWBRIDGE_PARACHAIN_KEY="//Relay"
-    export SNOWBRIDGE_RELAYCHAIN_KEY="//Alice"
-
-    build/snowbridge-relay run --config $configdir/config.toml >$logfile 2>&1 &
-
-    popd
-    echo "Relay PID: $!"
+    echo "Starting relay services"
+
+    # Build relay services
+    mage -d ../relayer build
+
+    # Configure beefy relay
+    jq \
+        --arg k1 "$(address_for BeefyLightClient)" \
+    '
+      .sink.contracts.BeefyLightClient = $k1
+    ' \
+    config/beefy-relay.json > $configdir/beefy-relay.json
+
+    # Configure parachain relay
+    jq \
+        --arg k1 "$(address_for BasicInboundChannel)" \
+        --arg k2 "$(address_for IncentivizedInboundChannel)" \
+        --arg k3 "$(address_for BeefyLightClient)" \
+    '
+      .source.contracts.BasicInboundChannel = $k1
+    | .source.contracts.IncentivizedInboundChannel = $k2
+    | .source.contracts.BeefyLightClient = $k3
+    | .sink.contracts.BasicInboundChannel = $k1
+    | .sink.contracts.IncentivizedInboundChannel = $k2
+    ' \
+    config/parachain-relay.json > $configdir/parachain-relay.json
+
+    # Configure ethereum relay
+    jq \
+        --arg k1 "$(address_for BasicOutboundChannel)" \
+        --arg k2 "$(address_for IncentivizedOutboundChannel)" \
+    '
+      .source.contracts.BasicOutboundChannel = $k1
+    | .source.contracts.IncentivizedOutboundChannel = $k2
+    ' \
+    config/ethereum-relay.json > $configdir/ethereum-relay.json
+
+    relay=$(realpath ../relayer/build/snowbridge-relay)
+
+    # Launch beefy relay
+    (
+        : > beefy-relay.log
+        while :
+        do
+            echo "Starting beefy relay"
+            ${relay} run beefy \
+                --config $configdir/beefy-relay.json \
+                --ethereum.private-key "0x935b65c833ced92c43ef9de6bff30703d941bd92a2637cb00cfad389f5862109" \
+                >>beefy-relay.log 2>&1 || true
+            sleep 20
```

Comment:
ah nice, so restart is handled automatically now by start-services? how is it handled in the staging/prod context?

Reply:
Yeah the start script will automatically restart the relay if it failes. In staging Nomad automatically restarts failed services, subject to the configuration you provide it.

For example, in the nomad output below, you can see the ethereum relay service failed 3 times, but still has 1 running instance - so its healthy.
```
$ nomad status ethereum-relay
ID            = ethereum-relay
Name          = ethereum-relay
Submit Date   = 2021-08-02T12:34:44Z
Type          = service
Priority      = 50
Datacenters   = dc1
Namespace     = default
Status        = running
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
main        0       0         1        3       0         0

Allocations
ID        Node ID   Task Group  Version  Desired  Status   Created     Modified
ad1b6f16  1d7b49bc  main        0        run      running  16m18s ago  16m5s ago
2c3c6887  1d7b49bc  main        0        stop     failed   5h6m ago    16m18s ago
```